includes/form-fields.php

```diff
@@ -27,16 +26,14 @@
         )
     );
     
-    if($possible_ignore_roles){
-        $this->form_fields['ignore_for_roles'] = array(
-            'title'             => __( 'Ignore tracking for roles', 'metrilo-analytics' ),
-            'type'              => 'multiselect',
-            'description'       => __( '<strong style="color: #999;">(Optional)</strong> If you check any of the roles, tracking data will be ignored for WP users with this role', 'metrilo-analytics' ),
-            'desc_tip'          => false,
-            'default'           => '',
-            'options'           => $possible_ignore_roles
-        );
-    }
+    $this->form_fields['ignore_for_roles'] = array(
+        'title'             => __( 'Ignore tracking for roles', 'metrilo-analytics' ),
+        'type'              => 'multiselect',
+        'description'       => __( '<strong style="color: #999;">(Optional)</strong> If you check any of the roles, tracking data will be ignored for WP users with this role', 'metrilo-analytics' ),
+        'desc_tip'          => false,
+        'default'           => '',
+        'options'           => $possible_ignore_roles
```

Comment:
If `is_admin()` returns `false`, you won't set `$possible_ignore_roles` . You should move this to the `if` here: https://github.com/Metrilo/metrilo-woocommerce-integration/pull/49/commits/7a94d3a73d1f27bd06f75e60534ca65628e9bff7#diff-8ac8470894c67c5a3d8d06aab27adf0929e9be34de0e4b23c44abe7578b950f7R4

Reply:
This should not be accessible out of the admin context, if somehow you manage to get to that page without admin privileges you will see empty options list []

Reply:
You are initializing the array inside the `if`: 
https://github.com/Metrilo/metrilo-woocommerce-integration/blob/7a94d3a73d1f27bd06f75e60534ca65628e9bff7/includes/form-fields.php#L6

and you use it here: https://github.com/Metrilo/metrilo-woocommerce-integration/blob/7a94d3a73d1f27bd06f75e60534ca65628e9bff7/includes/form-fields.php#L35

What will be the value of $possible_ignore_roles? `null`? If so, won't that be a problem for the code that uses the `options`?

Reply:
The file you are referencing was updated. At the moment $possible_ignore_roles is defined above that if conditional with default value of empty string. If the form-fields file reached without admin context (witch should not happen) then the options will have empty array (otherwise said no options to select from).

Reply:
True, my bad. Sorry. :rocket: